device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/twin/DirectMethod.java

```diff
@@ -64,7 +66,10 @@ public IotHubMessageResult execute(Message message, Object callbackContext)
                              **Codes_SRS_DEVICEMETHOD_25_008: [**If the message is of type DirectMethod and DEVICE_OPERATION_METHOD_RECEIVE_REQUEST then user registered device method callback gets invoked providing the user with method name and payload along with the user context. **]**
                              */
                             log.trace("Executing method invocation callback for method name {} for message {}", methodMessage.getMethodName(), methodMessage);
-                            DirectMethodResponse responseData = methodCallback.onMethodInvoked(methodMessage.getMethodName(), methodMessage.getBytes(), deviceMethodCallbackContext);
+
+                            MethodParser methodParser = new MethodParser();
+                            JsonElement jsonElement = methodParser.getPayloadFromJson(new String(methodMessage.getBytes()));
```

Comment:
if methodMessage payload is null, will this call throw?

Reply:
no, instead it will return a new JsonObject which should match your testing result.